source/log4net-loggly/LogglyAsyncHandler.cs

```diff
@@ -52,6 +53,14 @@ public void PostMessage(string msg, ILogglyAppenderConfig config)
 				Queue.TryAdd(msg);
 			}
 		}
+
+	    public void Flush()
+	    {
+	        while (IsRunning && Queue.Any())
```

Comment:
This does not guarantee that messages from queue are really sent. Message is taken from queue before it's sent so this condition can become `true` but messages are still not sent. The application can terminate before the actual HTTP request is made and data are delivered to Loggly.